framework/include/base/Registry.h

```diff
@@ -9,14 +9,22 @@
 
 #pragma once
 
+#include "InputParameters.h"
+
 #include <string>
 #include <vector>
 #include <set>
 #include <map>
+#include <memory>
 
 #define combineNames1(X, Y) X##Y
 #define combineNames(X, Y) combineNames1(X, Y)
 
+class InputParameters;
+InputParameters emptyInputParameters();
+template <class>
+InputParameters validParams();
```

Comment:
don't you dare

Reply:
That's not very informative. I think I'm going to have to include all of `InputParameters.h`. You call `callValidParams()`, whose definition is in `InputParameters.h`. Up to this point you've been getting lucky that `InputParameters.h` has been included wherever `Registry::add` has gotten instantiated, but that's no longer the case in https://civet.inl.gov/job/903311/ for instance. So if your objection is to the forward declarations, then yes those should indeed be removed

Reply:
Ah yeah this makes sense. I wouldn't object to moving `callValidParams()` elsewhere, but I can't really think of a better place for it to go so that it can be used in places other than just the registry.

In a few months this won't matter, #19440 removes `callValidParams()` seeing that we can simply call `&T::validParams` once the SFINAE crud goes away. Ok, so I'm fine with this unless you have a better option. Just request a change on #19440 about removing this

Reply:
Ok yea once this is merged, then I'll go add the deletion of the `InputParameters` include from #19440